cmd/traffic/cmd/manager/internal/ambassadoragent/cloudtoken/service.go

```diff
@@ -0,0 +1,121 @@
+package cloudtoken
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/datawire/dlib/dlog"
+	"github.com/datawire/k8sapi/pkg/k8sapi"
+	"github.com/telepresenceio/telepresence/v2/cmd/traffic/cmd/manager/managerutil"
+	apiv1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type Service interface {
+	MaybeAddToken(ctx context.Context, apikey string) error
+}
+
+type createConfigmapIfNotPresent struct {
+	createToken func(ctx context.Context, apikey string) error
```

Comment:
What motivates this field? Can't it just be a function?

Reply:
I feel like in the future we could present options, like `createConfigmapIfNotPresent` or `createSecretIfNotPresent`, which could both impl cloudToken.Service